scipy/signal/lti_conversion.py

```diff
@@ -400,21 +402,27 @@ def cont2discrete(system, dt, method="zoh", alpha=None):
         pp. 204-206, 1998.
 
     """
-    if len(system) == 1:
-        return system.to_discrete()
-    if len(system) == 2:
+    try:
+        n = len(system)
+    except TypeError:
+        return system.to_discrete(dt, method=method, alpha=alpha)
```

Comment:
I'd wait for a `signal` guru to review before making any changes, but since you've already drafted some tests that run locally it may be sensible to think about how they may be adjusted to work within our test suite to guard against regression in the future.

Reply:
OK, let's wait for some `signal` specialist. It will be easy for him/her to integrate a new check within your test suite (I have to take some time to look at how it is working).

Reply:
This will basically force everything to try a method of `to_discrete` for example if I enter 1, `len` will fail so it will try and give a very weird error `AttributeError: 'int' object has no attribute 'to_discrete'`

Reply:
Still not super nice but at least it should better test `if not isinstance(system, tuple)` etc. instead of this crystal-ball test. It is unfortunately matlabic notation but it is what it is currently.

Reply:
The test should test those bad inputs demonstrating the edge cases are out but the actual desired inputs are let in.